controllers/controll_plane_test.go

```diff
@@ -240,5 +263,254 @@ var _ = Describe("minimal", func() {
+						candidate.Spec.CommonName = "random"
+						candidate.Spec.IssuerRef = cmmetav1.ObjectReference{
+							Name: caIssuer.Name,
+							Kind: "ClusterIssuer",
+						}
+					})
+					randomCertName := randomCert
+
+					ctx := context.Background()
+
+					crd := createBaseCRD(defaultNamespace)
+
+					sharedOperandCertName := installOperandCert(crd.Name, defaultNamespace)
+
+					overrideSecretName := "control-plane-override"
+					if sharedOverride {
+						overrideSecretName = crd.Name + "-control-plane-override"
+					}
+					By("creating control-plane-override secret with custom cert-users and cert-roles")
+					overrideSecret := &corev1.Secret{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Secret",
+							APIVersion: "v1",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      overrideSecretName,
+							Namespace: defaultNamespace,
+						},
+						StringData: map[string]string{
+							"_cert-users": `# generated by test
+#
+					hawtio=/CN = hawtio-online\.hawtio\.svc.*/
+					operator=/.*activemq-artemis-operator.*/
+					probe=/.*activemq-artemis-operand.*/
+					new-user=/.*new-user.*/
+					`,
+							"_cert-roles": `# generated by test
+#
+					status=operator,probe
+					metrics=operator,new-user
+					hawtio=hawtio
+					`,
+						},
+					}
+					Expect(k8sClient.Create(ctx, overrideSecret)).Should(Succeed())
+					DeferCleanup(func() {
+						k8sClient.Delete(ctx, overrideSecret)
+					})
+
+					crd.Spec.Restricted = common.NewTrue()
+					crd.Spec.BrokerProperties = []string{
+						"messageCounterSamplePeriod=500",
+					}
+
+					By("Deploying the CRD " + crd.ObjectMeta.Name)
+					Expect(k8sClient.Create(ctx, &crd)).Should(Succeed())
+
+					brokerKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+					createdCrd := &brokerv1beta1.ActiveMQArtemis{}
+
+					waitForBrokerReady(ctx, k8sClient, brokerKey)
+
+					serverName := common.OrdinalFQDNS(crd.Name, defaultNamespace, 0)
+
+					By("Test 1: Operator cert can scrape metrics (default behavior)")
+					Eventually(func(g Gomega) {
+						transport := http.DefaultTransport.(*http.Transport).Clone()
+						httpClient := http.Client{
+							Transport: transport,
+							Timeout:   time.Second * 3,
+						}
+
+						httpClientTransport := httpClient.Transport.(*http.Transport)
+						httpClientTransport.TLSClientConfig = &tls.Config{
+							ServerName:         serverName,
+							InsecureSkipVerify: false,
+						}
+						httpClientTransport.TLSClientConfig.GetClientCertificate =
+							func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+								return common.GetOperatorClientCertificate(k8sClient, cri)
+							}
+
+						if rootCas, err := common.GetRootCAs(k8sClient); err == nil {
+							httpClientTransport.TLSClientConfig.RootCAs = rootCas
+						}
+
+						resp, err := httpClient.Get("https://" + serverName + ":8888/metrics")
+						fmt.Printf("Test 1 - Operator cert: status=%d, err=%v\n", resp.StatusCode, err)
```

Comment:
resp could be nil if err is not nil and cause a panic error getting `resp.StatusCode`